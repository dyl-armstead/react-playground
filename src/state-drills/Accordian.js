import React, { Component } from 'react';

export default class Accordion extends Component {
    static defaultProps = {
      sections: []
    };
  
    state = {
      activeSectionIndex: null,
    }
  
    handleSetActiveSecion = (sectionIndex) => {
      this.setState({ activeSectionIndex: sectionIndex })
    }
  
    renderItem(section, i, activeSectionIndex) {
      return (
        <li className='Accordion__item' key={i}>
          <button
            type='button'
            onClick={() => this.handleSetActiveSecion(i)}
          >
            {section.title}
          </button>
          {(activeSectionIndex === i) && <p>{section.content}</p>}
        </li>
      )
    }
  
    render() {
      const { activeSectionIndex } = this.state
      const { sections } = this.props
      return (
        <ul className='Accordion'>
          {sections.map((section, i) =>
            this.renderItem(section, i, activeSectionIndex)
          )}
        </ul>
      )
    }
  }

