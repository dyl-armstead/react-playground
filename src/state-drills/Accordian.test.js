import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordian from './Accordian';

const sectionsProp = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ]

describe(`Accordian Component`, () => {

      it('renders empty Accordian without errors', () => {
        const wrapper = shallow(<Accordian />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders empty Accordian by default', () => {
        const wrapper = shallow(<Accordian sections={sectionsProp}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders one Accordian section when clicked', () => {
        const wrapper = shallow(<Accordian sections={sectionsProp} />)
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders only last section after multiple section clicks', () => {
        const wrapper = shallow(<Accordian sections={sectionsProp} />)
        wrapper.find('button').at(1).simulate('click')
        wrapper.find('button').at(2).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})


