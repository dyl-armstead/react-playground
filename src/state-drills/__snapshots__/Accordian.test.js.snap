// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian Component renders empty Accordian by default 1`] = `
<ul
  className="Accordion"
>
  <li
    className="Accordion__item"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    className="Accordion__item"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
  </li>
  <li
    className="Accordion__item"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordian Component renders empty Accordian without errors 1`] = `
<ul
  className="Accordion"
/>
`;

exports[`Accordian Component renders one Accordian section when clicked 1`] = `
<ul
  className="Accordion"
>
  <li
    className="Accordion__item"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    className="Accordion__item"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    className="Accordion__item"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordian Component renders only last section after multiple section clicks 1`] = `
<ul
  className="Accordion"
>
  <li
    className="Accordion__item"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    className="Accordion__item"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
  </li>
  <li
    className="Accordion__item"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;
